### PR TITLE
Improve logging and behaviour of createJob and addNodes endpoints

### PR DESCRIPTION
**Pull request by mklkun**

### CODE REVIEW

**Review by AlexFalessi — APPROVED** [review]

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/JobService.java [hunk 1]
```diff
@@ -168,6 +173,7 @@ private String findTaskByProvidedPort(List<TaskDefinition> tasks, String provide
             if (taskProvidesPort(task, providedPortName))
                 return task.getName();
         }
+        LOGGER.error("Task that provides port " + providedPortName + " was not found in job.");
```
Comment (fviale) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/JobService.java]: you should add the job id to the error message

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java [hunk 2]
```diff
@@ -138,7 +149,11 @@ public synchronized Boolean addNodes(String sessionId, List<IaasDefinition> node
             LOGGER.info("Node source defined.");
 
             LOGGER.info("Trying to retrieve task: " + node.getTaskName());
-            Task task = repositoryService.getJob(jobId).findTask(node.getTaskName());
+            Task task = job.findTask(node.getTaskName());
+            if (task == null) {
+                LOGGER.error("Task [{}] does not exist.", node.getTaskName());
```
Comment (fviale) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java]: does the task name contain the job id ? if not, is it possible to also print the associated job id ?

**Review by fviale — APPROVED** [review]
approved, 2 comments

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/JobService.java (hunk 1 above)
Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/JobService.java]: Ok.

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java (hunk 2 above)
Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java]: No but I can add it.

**Pull request merged**